src/rsa.rs

```diff
@@ -338,4 +399,39 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn test_signature_rsa_pkcs1_sign() {
+        file_test::run("src/rsa_pkcs1_sign_tests.txt", |section, test_case| {
+            assert_eq!(section, "");
+
+            let digest_name = test_case.consume_string("Digest");
+            let alg = if digest_name == "SHA1" {
+                &digest::SHA1
+            } else if digest_name == "SHA256" {
+                &digest::SHA256
+            } else if digest_name == "SHA384" {
+                &digest::SHA384
+            } else if digest_name == "SHA512" {
+                &digest::SHA512
+            } else {
+                panic!("Unsupported digest: {}", digest_name);
+            };
+
+            let private_key = test_case.consume_bytes("Key");
+            let private_key = try!(untrusted::Input::new(&private_key));
+            let key_pair = RSAKeyPair::from_der(private_key).unwrap();
+
+            let msg = test_case.consume_bytes("Msg");
+            let expected = test_case.consume_bytes("Sig");
+            let mut actual: Vec<u8> = Vec::with_capacity(key_pair.public_modulus_len());
+            for _ in 0..actual.capacity() {
+                actual.push(0);
+            }
+            try!(key_pair.sign(alg, &msg, actual.as_mut_slice()));
+            assert_eq!(actual.as_slice(), &expected[..]);
```

Comment:
In another commit, we should add a check that the signature verifies with the public key. Note that currently we don't have the function to get the public key for the private key! We need to add that.